_posts/2016-04-01-new-post.md

```diff
@@ -0,0 +1,912 @@
+// app/mirage/config.js
+
+export default function() {
+  let comments = [
+    { id: 1, author: 'Yehuda Katz', text: 'This is one comment' },
+    { id: 2, author: 'Tom Dale', text: 'This is *another* comment' }
+  ];
+
+  this.get('/api/comments', () => {
+    return comments;
+  });
+
+  this.post('/api/comments', (db, request) => {
+    let comment = JSON.parse(request.requestBody);
+    comments = comments.concat([comment]);
+    return comments;
+  });
+}
+```
+
+#### Updating state
+
+Now that we have our mock server set up, let's start fetching some data.
+
+The first thing we're going to do is inject the [ember-ajax][todo] service
+into our component. We could use the usual `$.ajax` but the ember-ajax addon is
+included by default with ember-cli and provides a much nicer API.
+
+```javascript
+// app/components/comment-box.js
+
+import Ember from 'ember';
+
+const {
+  Component,
+  inject
+} = Ember;
+
+const { service } = inject;
+
+export default Component.extend({
+  ajax: service(),
+
+  init() {
+    this._super(...arguments);
+
+    this.data = [];
+  }
+});
+```
+
+Now we can use the `ajax` service to fetch our comment data when a `CommentBox`
+is created:
+
+```javascript
+// app/components/comment-box.js
+
+import Ember from 'ember';
+
+const {
+  Component,
+  get,
+  inject,
+  set
+} = Ember;
+
+const { service } = inject;
+
+export default Component.extend({
+  ajax: service(),
+
+  init() {
+    this._super(...arguments);
+
+    this.data = [];
+
+    get(this, 'ajax')
+      .request(get(this, 'url'))
+      .then((comments) => set(this, 'data', comments))
+      .catch((error) => console.error(error));
+  }
+});
+```
+
+Here we use the `ajax` service to send a request to the URL we've passed in as
+the `url` property of our component. When we receive a successful response, we
+set a new value for our component's `data` property, which will cause our
+component to rerender.
+
+Let's implement live updates using simple polling:
+
+```javascript
+// app/components/comment-box.js
+
+import Ember from 'ember';
+
+const {
+  Component,
+  get,
+  inject,
+  set
+} = Ember;
+
+const { service } = inject;
+
+export default Component.extend({
+  ajax: service(),
+
+  init() {
+    this._super(...arguments);
+
+    this.data = [];
+
+    this.loadCommentsFromServer();
+
+    setInterval(
+      this.loadCommentsFromServer.bind(this),
+      get(this, 'pollInterval')
+    );
+  },
+
+  loadCommentsFromServer() {
+    get(this, 'ajax')
+      .request(get(this, 'url'))
+      .then((comments) => set(this, 'data', comments))
+      .catch((error) => console.error(error));
+  }
+});
+```
+
+```handlebars
+{{!-- app/templates/application.hbs --}}
+
+{{comment-box url="/api/comments" pollInterval=2000}}
+```
+
+We moved the data fetching to a separate function and call it when our component
+is first initialized and every 2 seconds after that. Open your browser's console
+and you should see the network requests being logged by Mirage.
```

Comment:
It was strange to me that I didn't see any XHR requests at this point. Does Mirage hijack the ajax stuff?